server/src/main/java/org/elasticsearch/repositories/blobstore/ChecksumBlobStoreFormat.java

```diff
@@ -150,7 +150,7 @@ public void write(T obj, BlobContainer blobContainer, String name) throws IOExce
         final String blobName = blobName(name);
         writeTo(obj, blobName, bytesArray -> {
             try (InputStream stream = bytesArray.streamInput()) {
-                blobContainer.writeBlob(blobName, stream, bytesArray.length(), true);
+                blobContainer.writeBlob(blobName, stream, bytesArray.length(), false);
```

Comment:
This is idempotent as far as I can see. It seems easier to simply quietly skip an existing file than to suppress the file already exists exception and keep going with writing the index gen files upstream since S3 doesn't have this check anyway (the only downside I see is that we incur traffic for blob stores that do have a consistent existence check here, not sure if that's worth the extra effort though?)?

Reply:
The more I think about it, the more I dislike that we even have the `failIfAlreadyExists` flag on the `BlobContainer` interface. The parameter is simply ignored for S3 and won't reliably work with some NFS.
So in the end what we get is that our tests aren't really valid for S3 etc. in failover scenarios and the situation where any fixes we make to failover issues that are based on this check working will not be valid for S3.

=> I'm still a fan of this change, IMO it makes more sense fixing unforeseen issues with overwriting these files than fixing continuing to test and fix behavior that doesn't apply to all our blob stores.